26_Autoexposure/main.cpp

```diff
@@ -0,0 +1,874 @@
+			const IGPUCommandBuffer::SImageMemoryBarrier<IGPUCommandBuffer::SOwnershipTransferBarrier> imgBarriers1[] = {
+				{
+					.barrier = {
+						.dep = transferBarriers[0]
+						// no ownership transfers
+					},
+					.image = m_gpuImg.get(),
+					// transition the whole view
+					.subresourceRange = cpuImgParams.subresourceRange,
+					// a wiping transition
+					.newLayout = IGPUImage::LAYOUT::TRANSFER_DST_OPTIMAL
+				}
+			};
+			const IGPUCommandBuffer::SImageMemoryBarrier<IGPUCommandBuffer::SOwnershipTransferBarrier> imgBarriers2[] = {
+				{
+					.barrier = {
+						.dep = transferBarriers[1]
+						// no ownership transfers
+					},
+					.image = m_gpuImg.get(),
+					// transition the whole view
+					.subresourceRange = cpuImgParams.subresourceRange,
+					// a wiping transition
+					.oldLayout = IGPUImage::LAYOUT::TRANSFER_DST_OPTIMAL,
+					.newLayout = IGPUImage::LAYOUT::READ_ONLY_OPTIMAL
+				}
+			};
+			cmdbuf->pipelineBarrier(E_DEPENDENCY_FLAGS::EDF_NONE, { .imgBarriers = imgBarriers1 });
+			// upload contents
+			m_utils->updateImageViaStagingBuffer(
+				m_intendedSubmit,
+				cpuImgParams.image->getBuffer(),
+				cpuImgParams.image->getCreationParameters().format,
+				m_gpuImg.get(),
+				IGPUImage::LAYOUT::TRANSFER_DST_OPTIMAL,
+				cpuImgParams.image->getRegions()
+			);
+			cmdbuf->pipelineBarrier(E_DEPENDENCY_FLAGS::EDF_NONE, { .imgBarriers = imgBarriers2 });
+			m_utils->autoSubmit(m_intendedSubmit, [&](SIntendedSubmitInfo& nextSubmit) -> bool { return true; });
+
+			IGPUImageView::SCreationParams gpuImgViewParams = {
+				.image = m_gpuImg,
+				.viewType = IGPUImageView::ET_2D,
+				.format = m_gpuImg->getCreationParameters().format
+			};
+
+			m_gpuImgView = m_device->createImageView(std::move(gpuImgViewParams));
+
+			IGPUDescriptorSet::SDescriptorInfo info1 = {};
+			info1.info.image.imageLayout = IImage::LAYOUT::READ_ONLY_OPTIMAL;
+			info1.desc = m_gpuImgView;
+
+			IGPUDescriptorSet::SDescriptorInfo info2 = {};
+			info2.info.image.imageLayout = IImage::LAYOUT::READ_ONLY_OPTIMAL;
+			info2.desc = m_gpuImgView; // FIXME: temporarily pass in input image
+
+			IGPUDescriptorSet::SWriteDescriptorSet writeDescriptors[] = {
+				{
+					.dstSet = m_lumaPresentDS[0].get(),
+					.binding = 0,
+					.arrayElement = 0,
+					.count = 1,
+					.info = &info1
+				},
+				{
+					.dstSet = m_lumaPresentDS[1].get(),
+					.binding = 0,
+					.arrayElement = 0,
+					.count = 1,
+					.info = &info2
+				}
+			};
+
+			m_device->updateDescriptorSets(2, writeDescriptors, 0, nullptr);
+
+			queue->endCapture();
+		}
+
+		return true;
+	}
+
+	// We do a very simple thing, display an image and wait `DisplayImageMs` to show it
+	inline void workLoopBody() override
+	{
+		const uint32_t SubgroupSize = m_physicalDevice->getLimits().maxSubgroupSize;
+
+		// Luma Meter
+		{
+			auto queue = getComputeQueue();
+			auto cmdbuf = m_computeCmdBufs[0].get();
+			cmdbuf->reset(IGPUCommandBuffer::RESET_FLAGS::NONE);
```

Comment:
I'd start the example same as the others with the frames in flight (ex24 doesn't do FIF because it displays the same image for a long time), so proper round robining of multiple commandbuffers, blocks and waits before resets